utils/modelUtils.js

```diff
@@ -209,3 +223,12 @@ module.exports.pickRefData = function (modelName, item) {
 module.exports.resolveKey = function (name, key) {
   return getPKFieldType(name, key) === 'string' ? `'${key}'` : `${key}`
 }
+
+/**
+ * Check if model has dublicate field for the primary key
+ * @param {object} modelName
+ */
+module.exports.checkDublicate = function (modelName) {
+  const Model = Arrow.getModel(modelName)
+  return Model.metadata && Model.metadata['appc.odata'] ? Model.metadata['appc.odata'].primaryKey : false
```

Comment:
Originally, the pk key field is called 'primarykey', not 'primaryKey', fix it and check is it working.